pugl/pugl.h

```diff
@@ -218,7 +218,7 @@ typedef struct {
 	PuglEventType type;        /**< PUGL_BUTTON_PRESS or PUGL_BUTTON_RELEASE. */
 	PuglView*     view;        /**< View that received this event. */
 	uint32_t      flags;       /**< Bitwise OR of PuglEventFlag values. */
-	uint32_t      time;        /**< Time in milliseconds. */
+	double        time;        /**< Time in seconds. */
```

Comment:
This gives me pause... I agree that seconds is the best unit in general, but worry a bit about the imprecision of floating point.  Could not being able to get back to the system-provided integer value possibly be a problem?

I suppose pugl code should stay in "pugl land", so probably not.

Reply:
I can't think of a case where you would want to get the system integer back. At least the way I use the timestamps, what matters is the difference of two timestamps, and if we alllow doubles, then fractional seconds are the more natural representation.

If you do want the system provided integer back, you can multiply with 1000 and round to the nearest integer value. Since doubles have 52 bits for representing the "number part", this should be perfectly accurate for 32bit integer timestamps. For 64bit integer timestamps, it should still be accurate if the timestamp is not too large (assuming it initially starts at zero it should never be too large).

Reply:
Fair enough.  This struct is a blatant rip off of the X one (which means it also almost exactly matches the Gtk one), but I can't think of a particularly good reason not to do this here either.